```
/var/lib/jenkins/jobs/dron/workspace/Nodes/master/build/fw_sparky/fw_sparky.elf section `.data' will not fit in region `flash'
/home/mlyle/dronin/tools/gcc-arm-none-eabi-4_9-2015q1/bin/../lib/gcc/arm-none-eabi/4.9.3/../../../../arm-none-eabi/bin/ld: region `flash' overflowed by 48 bytes
collect2: error: ld returned 1 exit status
```

Wow, looks like we're really down to the wire on sparky again.

--- a/flight/targets/aq32/fw/pios_board.c
+++ b/flight/targets/aq32/fw/pios_board.c
@@ -210,14 +210,6 @@ void PIOS_Board_Init(void) {
     PIOS_LED_Init(led_cfg);
 #endif    /* PIOS_INCLUDE_LED */
 
-#if defined(PIOS_INCLUDE_I2C)
-    if (PIOS_I2C_Init(&pios_i2c_internal_id, &pios_i2c_internal_cfg)) {
-        PIOS_DEBUG_Assert(0);
-    }
-    if (PIOS_I2C_CheckClear(pios_i2c_internal_id) != 0)
-        panic(8);
-#endif
-
 #if defined(PIOS_INCLUDE_SPI)
     if (PIOS_SPI_Init(&pios_spi_internal_id, &pios_spi_internal_cfg)) {
         PIOS_Assert(0);
@@ -381,6 +373,23 @@ void PIOS_Board_Init(void) {
 #endif    /* PIOS_INCLUDE_USB */
 
     /* Configure the IO ports */
+
+    PIOS_HAL_ConfigurePort(HWSHARED_PORTTYPES_I2C,  // port type protocol
+            NULL,                                   // usart_port_cfg
+            NULL,                                   // frsky usart_port_cfg
+            NULL,                                   // com_driver
+            &pios_i2c_internal_id,                  // i2c_id
+            &pios_i2c_internal_cfg,                 // i2c_cfg
+            NULL,                                   // ppm_cfg
+            NULL,                                   // pwm_cfg
+            PIOS_LED_ALARM,                         // led_id
+            NULL,                                   // usart_dsm_hsum_cfg
+            NULL,                                   // dsm_cfg
+            0,                                      // dsm_mode
+            NULL,                                   // sbus_rcvr_cfg
+            NULL,                                   // sbus_cfg    
+            false);                                 // sbus_toggle
+
     HwAQ32DSMxModeOptions hw_DSMxMode;
     HwAQ32DSMxModeGet(&hw_DSMxMode);
 
@@ -623,15 +632,21 @@ void PIOS_Board_Init(void) {
 
     if (Magnetometer == HWAQ32_MAGNETOMETER_EXTERNAL)
     {
-        AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_OK);
-            
-        if (PIOS_I2C_Init(&pios_i2c_external_id, &pios_i2c_external_cfg)) {
-            PIOS_DEBUG_Assert(0);
-            AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
-        }
-        
-        if (PIOS_I2C_CheckClear(pios_i2c_external_id) != 0)
-            AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);;
+		PIOS_HAL_ConfigurePort(HWSHARED_PORTTYPES_I2C,  // port type protocol
+				NULL,                                   // usart_port_cfg
+				NULL,                                   // frsky usart_port_cfg
+				NULL,                                   // com_driver
+				&pios_i2c_external_id,                  // i2c_id
+				&pios_i2c_external_cfg,                 // i2c_cfg
+				NULL,                                   // ppm_cfg
+				NULL,                                   // pwm_cfg
+				PIOS_LED_ALARM,                         // led_id
+				NULL,                                   // usart_dsm_hsum_cfg
+				NULL,                                   // dsm_cfg
+				0,                                      // dsm_mode
+				NULL,                                   // sbus_rcvr_cfg
+				NULL,                                   // sbus_cfg    
+				false);                                 // sbus_toggle
     
         if (PIOS_HMC5883_Init(pios_i2c_external_id, &pios_hmc5883_external_cfg) == 0) {
             if (PIOS_HMC5883_Test() == 0) {

--- a/flight/targets/aq32/fw/pios_config.h
+++ b/flight/targets/aq32/fw/pios_config.h
@@ -125,7 +125,8 @@
 #define IDLE_COUNTS_PER_SEC_AT_NO_LOAD (9873737)
 
 #define REVOLUTION
-#define AQ32
+
+#define SUPPORTS_EXTERNAL_MAG
 
 #endif /* PIOS_CONFIG_H */
 /**

--- a/shared/uavobjectdefinition/hwaq32.xml
+++ b/shared/uavobjectdefinition/hwaq32.xml
@@ -116,7 +116,7 @@
 		<field name="MPU6000Rate" units="" type="enum" elements="1" options="200,333,500,666,1000,2000,4000,8000" defaultvalue="500"/>
 		<field name="MPU6000DLPF" units="" type="enum" elements="1" options="256,188,98,42,20,10,5" defaultvalue="188"/>
 	
-		<field name="Magnetometer" units="function" type="enum" elements="1" options="Internal,External" defaultvalue="Internal"/>
+		<field name="Magnetometer" units="function" type="enum" elements="1" options="Disabled,Internal,External" defaultvalue="Internal"/>
 		<field name="ExtMagOrientation" units="function" type="enum" elements="1" options="Top0degCW,Top90degCW,Top180degCW,Top270degCW,Bottom0degCW,Bottom90degCW,Bottom180degCW,Bottom270degCW" defaultvalue="Top0degCW" />
 	
 		<access gcs="readwrite" flight="readwrite"/>

